agent/host_agent_test.go

```diff
@@ -71,53 +77,90 @@ var _ = Describe("Agent", func() {
 			}
 			Expect(k8sClient.Create(context.TODO(), byoHost)).NotTo(HaveOccurred())
 
-			command := exec.Command(pathToHostAgentBinary, "--kubeconfig", kubeconfigFile.Name(), "--namespace", ns.Name, "--downloadpath", fakeDownloadPath)
-			session, err = gexec.Start(command, GinkgoWriter, GinkgoWriter)
+			runner.CommandArgs["--downloadpath"] = fakeDownloadPath
+			output, _, err := runner.ExecByoDockerHost(byoHostContainer)
 			Expect(err).NotTo(HaveOccurred())
-			Consistently(session).ShouldNot(gexec.Exit(0))
+
+			defer output.Close()
+			f := e2e.WriteDockerLog(output, agentLogFile)
+			defer func() {
+				deferredErr := f.Close()
+				if deferredErr != nil {
+					e2e.Showf("error closing file %s: %v", agentLogFile, deferredErr)
+				}
+			}()
+			Consistently(func() (done bool) {
+				_, err := os.Stat(agentLogFile)
+				if err == nil {
+					data, err := os.ReadFile(agentLogFile)
+					if err == nil && strings.Contains(string(data), "\"msg\"=\"error\"") {
+						return true
+					}
+				}
+				return false
+			}, 10).Should(BeFalse())
```

Comment:
Do we need `10s` here? Its not a time consuming operation on the agent side, right? The default timeout for `Consistently` should be enough I guess